packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -407,7 +410,12 @@ export class HttpPlugin extends BasePlugin<Http> {
       const span = plugin._startHttpSpan(operationName, spanOptions);
       plugin._tracer
         .getHttpTextFormat()
-        .inject(span.context(), Format.HTTP, options.headers);
+        // Using context directly like this is temporary. In a future PR, context
+        // will be managed by the scope manager (which may be renamed to context manager?)
+        .inject(
+          Context.setActiveSpan(Context.ROOT_CONTEXT, span),
+          options.headers!
```

Comment:
In this case the headers could not be undefined based on the code here: https://github.com/open-telemetry/opentelemetry-js/blob/master/packages/opentelemetry-plugin-http/src/utils.ts#L269

But the typings would definitely allow you to remove that code and break this. I'll fix the typings so that it will complain if you don't include headers and remove this.